.github/workflows/install_dev_env/action.yml

```diff
@@ -17,7 +17,7 @@ runs:
     - name: Installing poetry
       uses: abatilo/actions-poetry@v2.0.0
       with:
-        poetry-version: '1.2.2'
+        poetry-version: '1.7.1'
```

Comment:
Poetry 1.7.0 has added support for Python 3.12. 
See: https://python-poetry.org/blog/announcing-poetry-1.7.0/#official-support-for-python-312